sormas-ui/src/main/java/de/symeda/sormas/ui/vaccination/VaccinationController.java

```diff
@@ -129,6 +135,10 @@ public void edit(
 			createComponent.addDeleteWithReasonListener((deleteDetails) -> {
 				popupWindow.close();
 				if (doSave) {
+					List<CaseDataDto> cases = getCaseFacade().getRelevantCasesForVaccination(vaccination);
```

Comment:
```suggestion
List<CaseDataDto> cases = getCaseFacade().getRelevantCasesForVaccination(vaccination).filter(c -> getCasesFacade().hasOtherValidVaccination(c, vaccination.getUuid())).collect(Collectors.toList());
```

And then use this list for all the other stuff that is done afterwards (which of course also means that you don't need to check for valid vaccinations again in the next line, the `isEmpty` check is sufficient then).